chore: add IOT image support to audit script

HastD opened the pull request:
Add `Image.IOT` variant for `iot-*` images, and exempt IOT images from Trivalent DNS check (same as CoreOS images). Also improve parsing in `Image.from_image_ref` and fix some style issues flagged by Ruff.

RoyalOughtness commented on files/system/usr/libexec/secureblue/utils/__init__.py:
```diff
@@ -185,20 +185,23 @@ class Image(enum.Enum):
+        if "securecore-" in image_ref:
             return cls.COREOS
+        if "iot-" in image_ref:
+            return cls.IOT
```
ruff is mad at this additional return but IMO we can just comment to bypass or loosen the settings

HastD replied:
I rewrote it to have more precise parsing anyway, now it's not an issue.

alexvojproc reviewed (approved):
the new syntax is tidier - I forgot you'd told me about this :)

HastD committed:
chore: add IOT image support to audit script

Add `Image.IOT` variant for `iot-*` images, exempt IOT images from
Trivalent DNS check (same as CoreOS images), fix ruff issues, and make
Image.from_image_ref more precise.

Signed-off-by: Daniel Hast <hast.daniel@protonmail.com>

HastD commented:
Closing this since it got merged into #1493.
